cli/main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"math"
+	"os"
+	"strconv"
+	"time"
+
+	"context"
+
+	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/onchain"
+	"github.com/attestantio/go-eth2-client/spec"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	eigenpodproofs "github.com/Layr-Labs/eigenpod-proofs-generation"
+)
+
+type ValidatorWithIndex = struct {
+	Validator *phase0.Validator
+	Index     uint64
+}
+
+func main() {
+	eigenpod := flag.String("eigenpod", "", "[required] The onchain address of your eigenpod contract (0x123123123123)")
+	beacon := flag.String("beacon", "", "[required] URI to a functioning beacon node RPC (https://)")
+	node := flag.String("node", "", "[required] URI to a functioning execution-layer RPC")
+	out := flag.String("output", "", "Output path for the proof. (defaults to stdout)")
+	help := flag.Bool("help", false, "Prints the help message and exits.")
+
+	flag.Parse()
+
+	if help != nil && *help {
+		flag.Usage()
+		log.Fatal("Showing help.")
+	}
+
+	if *eigenpod == "" || *beacon == "" || *node == "" {
+		flag.Usage()
+		log.Fatal("Must specify: -eigenpod, -beacon, and -node.")
+	}
+
+	ctx := context.Background()
+
+	execute(ctx, *eigenpod, *beacon, *node, out)
+}
+
+func getBeaconClient(beacon_uri string) (BeaconClient, error) {
+	beaconClient, _, err := NewBeaconClient(beacon_uri)
+	return beaconClient, err
+}
+
+func lastCheckpointedForEigenpod(eigenpodAddress string, client *ethclient.Client) uint64 {
+	eigenPod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), client)
+	PanicOnError(err)
+
+	timestamp, err := eigenPod.LastCheckpointTimestamp(nil)
+	PanicOnError(err)
+	return timestamp
+}
+
+func computeSlotImmediatelyPriorToTimestamp(timestampSeconds uint64, genesis time.Time) uint64 {
+	return uint64(math.Floor(float64(timestampSeconds)-float64(genesis.Unix())) / 12)
+}
+
+// search through beacon state for validators whose withdrawal address is set to eigenpod.
+func findAllValidatorsForEigenpod(eigenpodPubKey string, beaconState *spec.VersionedBeaconState) []ValidatorWithIndex {
+	allValidators, err := beaconState.Validators()
+	PanicOnError(err)
+
+	expectedCredentials := sha256.Sum256([]byte(eigenpodPubKey))
+	expectedCredentials[0] = 0 // the first byte of withdrawal credentials is set to 0.
```

Comment:
so actually the only check we should be doing is that the last 20 bytes are the same as the eigenPodAddress. that will account for current and future withdrawal credential types

fyi current withdrawal credential types are 

https://github.com/Layr-Labs/eigenlayer-contracts/blob/d148952a2942a97a218a2ab70f9b9f1792796081/src/contracts/pods/EigenPod.sol#L663

which is 32 bytes:
1. starting with a 0x01 byte
2. next 11 0x00 bytes
3. then the address bytes

there will be a new credential type in pectra which replaces the 0x01 byte with a 0x02 byte.